lib/components/cluster-build/ClusterOptionsForm.js

```diff
@@ -29,23 +30,16 @@ class ClusterOptionsForm extends React.Component {
   onPlanChange = e => {
     const state = copy(this.state)
     state.selectedPlan = this.props.plans.find(p => p.metadata.name === e.target.value)
-    this.setState(state)
-  }
-
-  matchingClusters = (clusterName) => this.props.teamClusters.filter(tc => tc.metadata.name.indexOf(clusterName) >= 0)
-
-  rawClusterName = () => this.state.selectedPlan && this.state.selectedProvider ?
-    `${this.state.selectedProvider.metadata.name}-${this.props.team.metadata.name}-${this.state.selectedPlan.metadata.name}` :
-    ''
-
-  generateClusterName = () => {
-    let clusterName = this.rawClusterName()
-    const matchingClusters = this.matchingClusters(clusterName)
-    const matchingClusterCount = matchingClusters.length
-    if (matchingClusterCount) {
-      clusterName = `${clusterName}-${matchingClusterCount + 1}`
+    let clusterName = `${this.props.team.metadata.name}-${state.selectedPlan.metadata.name}`
+    const matchingCluster = this.props.teamClusters.find(tc => tc.metadata.name === clusterName)
+    if (matchingCluster) {
+      clusterName = `${clusterName}-2`
```

Comment:
Should this be in a loop in case cluster-2 already exists?

Reply:
It should never let a cluster with the same name be created, so worse case you would get `cluster-2-2` 😂